dbms/src/Common/MemoryTracker.cpp

```diff
@@ -17,6 +17,7 @@ namespace DB
 
 
 static constexpr size_t log_peak_memory_usage_every = 1ULL << 30;
+static constexpr Int64 untracked_memory_limit = 4 * 1024 * 1024;
```

Comment:
What is untracked memory? Is it to avoid frequent concurrent updates of the parent counter? This needs a comment.

I'd write it in terms of two counters: the current memory usage, and the usage we last passed to the parent tracker. If the difference is high, pass it to the parent, and update the last passed value.

Reply:
Untracked memory is memory that is allocated in new but could be not counted in MemoryTracker. Yes, it's a performance issue not to access to atomics. A comment added.